pallets/auctions/src/lib.rs

```diff
@@ -44,16 +42,114 @@ const BID_ADD_BLOCKS: u32 = 10;
 /// Minimal auction duration
 const MIN_AUCTION_DUR: u32 = 10;
 
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
+pub enum Auction<T: Config> {
```

Comment:
I'd suggest not to use Config here. The enum or the data structs dont really need to pallet's config implemetntion. They just need types.

Reply:
I think I need it for BalanceOf<T> later on? Otherwise it complains about missing generics

Reply:
You can easily use Balance generic.  

Eg. 

```suggestion
pub enum Auction<Balance> {
```

You would need to enumerate all types needed for the general data. 

But I still think it is a better approach.